application/uitests/org.openjdk.jmc.test.jemmy/src/main/java/org/openjdk/jmc/test/jemmy/misc/wrappers/JfrWizard.java

```diff
@@ -446,6 +455,22 @@ private boolean setEventSetting(Boolean desiredState, String settingName, String
 		return currentState;
 	}
 
+	public void setEventSettingInvertOthers(boolean desiredState, String settingName, String ... path) {
+		// invert everything
+		// the actual names are observed to always at least contain the roots of the tree
+		// and setting them also sets all their children
+		{
+			final var eventSettingsTree = getEventSettingsTree();
+			final var names = eventSettingsTree.getAllItemTexts();
+			for (var name : names) {
```

Comment:
You could simply do `for (var name : getEventSettingsTree().getAllItemTexts()) {` here

Reply:
Yes I know. There was a version of this where I first also used the eventSettingsTree to select the path. But I removed that because it turned out to not be needed.

Reply:
other than that I really struggle to find a reason to do it one way or the other, other than someone saying "no really, please do it this way" :)

Reply:
Sure, it's a matter of taste. :)